test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs

```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
                         string.Empty
                     },
                     {
+                        (Expression<Func<TestModel, int>>)(model => models[0].SelectedCategory.CategoryId),
+                        "models[0].SelectedCategory.CategoryId"
+                    },
+                    {
+                        (Expression<Func<TestModel, string>>)(model => modelTest.Name),
+                        "modelTest.Name"
+                    },
+                    {
+                        (Expression<Func<TestModel, Type>>)(model => modelType),
+                        "modelType"
+                    },
+                    {
```

Comment:
What happens if the expression uses something from a type or namespace called "Model" e.g `m => MyApp.Model.Constants.CopyrightYear` or some weird variant like this.